packages/frontend/src/views/CalendarView/CalendarView.tsx

```diff
@@ -0,0 +1,268 @@
+
+type Availability = {
+  start: Date;
+  end: Date;
+  available: boolean;
+};
+
+type Calendar = {
+  name: string;
+  id: string;
+  enabled: boolean;
+  onEnabledChanged: (enabled: boolean) => void;
+};
+
+type Participant = {
+  id: string;
+  name: string;
+};
+
+type FlockParams = {
+  flockCode: string;
+};
+
+const FIFTEEN_MINUTES = 15 * 60000;
+
+type FlockProps = {
+  flockName: string;
+  datesPicked: Date[];
+  timeRange: [Date, Date];
+  userAvailability: Availability[];
+  othersAvailability: Availability[];
+};
+
+const Flock: React.FC<FlockProps> = ({ flockName, datesPicked, timeRange, userAvailability, othersAvailability }) => {
+  return (
+    <TitleLayout
+      title={flockName}
+      content={
+        <>
+          <div className={styles.timeMatcher}>
+            <Timematcher
+              datesPicked={datesPicked}
+              timeRange={timeRange}
+              userAvailability={userAvailability}
+              othersAvailability={othersAvailability}
+            />
+            <Legend />
+          </div>
+        </>
+      }
+    />
+  );
+};
+
+const Legend: React.FC = () => {
+  return (
+    <div>
+      <h2>Legend</h2>
+      <div className={styles.legendKeys}>
+        <div className={styles.circleBothAvailable} />
+        <h3>All available</h3>
+        <div className={styles.circleUserAvailable} />
+        <h3>You're available</h3>
+        <div className={styles.circleFlockAvailable} />
+        <h3>Others available</h3>
+      </div>
+    </div>
+  );
+};
+
+const CalendarView: React.FC = () => {
+  const { flockCode } = useParams<FlockParams>();
+
+  const flock = useQuery<GetCurrentFlockResult>(GET_USER_FLOCK, {
+    variables: { flockCode: flockCode },
+  });
+
+  const participants = useQuery<GetCurrentFlockResult>(GET_FLOCK_PARTICIPANTS, {
+    variables: { flockCode: flockCode },
+  });
+  const calendars = useQuery<GetCurrentUserResult>(GET_USER_CALENDARS);
+
+  let participantList: Participant[] = [];
+  if (participants.data) {
+    const { users } = participants.data.getFlockByCode;
+    users.forEach((user) => {
+      const { id, name } = user;
+      participantList.push({ id, name });
+    });
+  }
+
+  let calendarList: Calendar[] = [];
+  let availabilityIds: string[] = [];
+  if (calendars.data) {
+    const { flocks, availability } = calendars.data.getCurrentUser;
+    const userAvailabilityForFlock = flocks.filter((flock) => flock.flockCode === flockCode)[0].userFlockAvailability;
+
+    availability.forEach((availability) => {
+      const { id, name } = availability as UserAvailabilityPartialDTO;
+      let isEnabled = false;
+      const availabilityForFlock = userAvailabilityForFlock.filter(
+        (userAvailability) => userAvailability.userAvailability === availability,
+      );
+      if (availabilityForFlock) {
+        isEnabled = true;
+      }
+      calendarList.push({ name, id, enabled: isEnabled, onEnabledChanged: () => {} });
```

Comment:
Shouldn't `onEnabledChanged` call a mutation query to set the enablement of the calendar for the flock?

Reply:
Fixed!!